### PR TITLE
Do not prevent default handling for keydown event except for backspac…

### PR DESCRIPTION
**Pull request by awtcode**
…e and tab.

### CODE REVIEW

**Comment by awtcode** [comment]
@kripken , I have deviated slightly from SDL2's implementation by preventing default handling even when text inputs are not enabled. This is to cater for the case whereby users might want to send keypress events to the HTML canvas.

**Comment by awtcode** [comment]
@kripken , did you get the chance to look at this?

**Comment by awtcode** [comment]
@kripken , this has been pending for the past 9 days so can I have your comments on this and whether it is ready for merging? Thanks.

File: src/proxyClient.js [hunk 1]
```diff
@@ -237,10 +237,24 @@ function cloneObject(event) {
...
+    return true; // NO keypress, NO back navigation
+  }
+};
+
 ['keydown', 'keyup', 'keypress', 'blur', 'visibilitychange'].forEach(function(event) {
   document.addEventListener(event, function(event) {
     worker.postMessage({ target: 'document', event: cloneObject(event) });
-    event.preventDefault();
+    
+    var preventDefault = shouldPreventDefault(event);
```
Comment (kripken) [inline src/proxyClient.js]: simpler to just write `if (shouldPreventDefault(event)) {`

**Comment by kripken** [comment]
Thanks, looks good except for one comment.

Sorry for the delay, I was on a trip.

**Comment by awtcode** [comment]
Thanks for your comment @kripken , I have already updated the pull request with the rework.

**Comment by kripken** [comment]
Thanks!

**Pull request merged**